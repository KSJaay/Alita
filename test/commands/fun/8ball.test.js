const {Client} = require("discord.js");
const eightballCommand = require("../../../commands/fun/8ball");
const logger = require("../../../logger");

jest.mock("discord.js");
jest.mock("../../../logger");

describe("8ball command", () => {
  const client = new Client();

  let interaction;

  beforeEach(() => {
    interaction = {
      options: {
        getSubcommand: jest.fn(),
        getChannel: jest.fn(),
        getString: jest.fn(),
      },
      guildId: "123456789",
      reply: jest.fn(),
      guild: {
        channels: {
          cache: {
            get: jest.fn().mockReturnValue({send: jest.fn()}),
          },
        },
      },
      user: {
        id: "987654321",
        username: "test",
      },
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should reply with a success message", async () => {
    await eightballCommand.execute(client, interaction, {});

    expect(interaction.reply).toHaveBeenCalledWith(
      expect.objectContaining({
        embeds: [expect.any(Object)],
        ephemeral: false,
      })
    );
  });

  it("should call logger.error if an error is thrown", async () => {
    interaction.reply.mockImplementation(() => {
      throw new Error("test");
    });

    await eightballCommand.execute(client, interaction, {});

    expect(logger.error).toHaveBeenCalledWith(
      `Error executing '8ball' command!`,
      {
        label: "Command",
        message: "test",
        data: {},
      }
    );
  });

  it("snaphot of interaction data", () => {
    expect(eightballCommand.interaction).toMatchSnapshot();
  });
});
