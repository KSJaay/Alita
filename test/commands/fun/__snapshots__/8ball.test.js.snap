// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currentbans command snaphot of interaction data 1`] = `
{
  "description": "Ask the magic 8 ball a question.",
  "name": "8ball",
  "options": [
    {
      "description": "What is your question?",
      "name": "question",
      "required": true,
      "type": 3,
    },
  ],
}
`;
